text/0707-modernize-built-in-components-2.md

```diff
@@ -0,0 +1,433 @@
+callbacks that would have been clobbered by these named arguments (may not be a
+complete list):
+
+* `@change`
+* `@focusIn`
+* `@focusOut`
+* `@keyDown`
+* `@keyPress`
+* `@keyUp`
+
+Passing these named arguments historically supressed certain behavior of the
+built-in components, in some cases preventing the components from functioning
+properly. This was never an intended part of the original design and should be
+considered a bug.
+
+This bug may be fixed at any time during the transition period – the supression
+behavior may stop without notice and should not be relied upon. An invocation
+with these named arguemnts shall trigger a deprecation warning with this
+additional caveat, similar to this:
+
+```
+<Input @change={{this.onChange}} />
+       ~~~~~~~~~~~~~~~~~~~~~~~~~
+or
+
+{{input change=this.onChange}}
+        ~~~~~~~~~~~~~~~~~~~~
+
+Passing the `@change` argument to <Input> is deprecated. This would have
+overwritten the internal `change` method on the <Input> component and prevented
+it from functioning properly. Instead, please use the {{on}} modifier, i.e.
+`<Input {{on "change" ...}} />` instead of `<Input @change={{...}} />` or
+`{{input change=...}}`.
+```
+
+### No Longer Recommended
+
+#### Changing `@tagName` on `<LinkTo>`
+
+Due to the classic component implementation heritage, the built-in components
+historically accepted a `@tagName` argument that allows customizing the tag
+name of the underlying HTML element.
+
+This was once popular with the `<LinkTo>` component for adding navigation
+behavior to buttons, table row and other UI elements. The current concensus is
+that this is an anti-pattern and causes issues with assistive technologies.
+
+In most cases, the `<a>` anchor HTML element should be used for navigational UI
+elements and styled with CSS to fit with the design requirements. Ocasionally,
+a button may be acceptable, in which case a custom event handler can be written
+using the router service and attached using the `{{on}}` modifier.
+
+Other edge cases exists, but generally those solutions can be adapted to fufill
+the requirements. For example, to make a table row clickable as a convenience,
+the primary column can be made into a link, while a click event handler is
+attached to the table row to redispatch the click to trigger the link.
+
+Since this feature is no longer recommended, invoking `<LinkTo>` with the
+`@tagName` argument shall trigger a deprecation warning similar to this:
+
+```
+<LinkTo @tagName="div" ...>...</LinkTo>
+        ~~~~~~~~~~~~~~
+or
+
+{{#link-to tagName="div" ...}}...{{/link-to}}
+           ~~~~~~~~~~~~~
+
+Passing the `@tagName` argument to <LinkTo> is deprecated. Using a <div>
+element for navigation is not recommended as it creates issues with assistive
+technologies. Remove this argument to use the default <a> element. In the rare
+cases that calls for using a different element, refactor to use the router
+service inside a custom event handler instead.
```

Comment:
Does it make sense to have a link here to appropriate section of the RRF, to provide more context, and detailed description like it is in the RFC.

This "no longer recommended" feels like a soft form of deprecations. But deprecations typically have their link.